docs/source/API/core/view/realloc.md

```diff
@@ -151,13 +157,23 @@ void realloc(const Impl::ViewCtorProp<ViewCtorArgs...>& arg_prop,
          const typename Kokkos::View<T, P...>::array_layout& layout);
   ```
   Resizes `v` to have the new dimensions without preserving its contents. The new `Kokkos::View` is constructed using the View constructor properties `arg_prop`, e.g., `Kokkos::view_alloc(Kokkos::WithoutInitializing)`.
+  May not modify `v` if the dimensions already match.
   * `v`: existing view, can be a default constructed one.
   * `layout`: a layout instance containing the new dimensions.
   * `arg_prop`: View constructor properties, e.g., `Kokkos::view_alloc(Kokkos::WithoutInitializing)`.
 
   Restrictions:
   * `arg_prop` must not include a pointer to memory, a label, or a memory space.
 
+## Possibly Unexpected Behavior Warning
+
+`realloc` will only modify the specific `View` instance passed to it.
+Any other `View` which aliases the same allocation will be unmodified.
+Consequently, if the `use_count()` of the `View` is larger than 1, the
+old allocation will not be deleted.
+Note that if the size arguments already match the extents of the `View`
+argument, that `realloc` may not create a new `View`.
```

Comment:
Did you consider having a code snippet to illustrate how this may have potentially surprising behavior?
Also should we spell out that it is best to think of if as what `std::realloc` is to raw pointers, that is the original view is "invalidated" and that keep using another object that aliases that same original view is a bad idea.